selfdrive/car/volkswagen/carstate.py

```diff
@@ -42,8 +45,16 @@ def update(self, pt_cp):
     ret.brakeLights = bool(pt_cp.vl["ESP_05"]['ESP_Status_Bremsdruck'])
 
     # Update gear and/or clutch position data.
-    can_gear_shifter = int(pt_cp.vl["Getriebe_11"]['GE_Fahrstufe'])
-    ret.gearShifter = self.parse_gear_shifter(self.shifter_values.get(can_gear_shifter, None))
+    if trans_type == TRANS.automatic:
+      ret.gearShifter = self.parse_gear_shifter(self.shifter_values.get(pt_cp.vl["Getriebe_11"]['GE_Fahrstufe'], None))
+    elif trans_type == TRANS.direct:
+      ret.gearShifter = self.parse_gear_shifter(self.shifter_values.get(pt_cp.vl["EV_Gearshift"]['GearPosition'], None))
+    elif trans_type == TRANS.manual:
+      ret.clutchPressed = not pt_cp.vl["Motor_14"]['MO_Kuppl_schalter']
+      if bool(pt_cp.vl["Gateway_72"]['BCM1_Rueckfahrlicht_Schalter']):
```

Comment:
Is there a way to get reverse signal from the gearbox directly instead of using the light?

Reply:
In the case of a manual trans, there's no gearbox controller to get it from. There's a contact switch in the shifter for Reverse wired to the BCM, and a clutch pedal contact switch wired to the ECM. That's about it.

In this case, I believe the signal is truly referring to the switch and not the light on the back of the car. "Rueckfahrlicht schalter" means "reversing light switch". To put that in context, VW refers to the pair of redundant brake pedal contacts as the  "brake light switch" and the "brake test switch", and those are both fully distinct from the tail braking lights being activated, which we can see happen by watching BCM messaging when ACC is decelerating without driver input.

Reply:
Ok, sounds good! The "switch" explanation makes sense to me.